src/main/java/uk/gov/pay/directdebit/notifications/clients/ClientFactory.java

```diff
@@ -1,78 +1,46 @@
 package uk.gov.pay.directdebit.notifications.clients;
 
-import io.dropwizard.client.JerseyClientBuilder;
-import io.dropwizard.client.JerseyClientConfiguration;
 import io.dropwizard.setup.Environment;
-import org.apache.http.config.RegistryBuilder;
-import org.apache.http.conn.HttpClientConnectionManager;
-import org.apache.http.conn.socket.ConnectionSocketFactory;
-import org.apache.http.conn.socket.PlainConnectionSocketFactory;
-import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
-import org.apache.http.impl.conn.ManagedHttpClientConnectionFactory;
-import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
-import org.glassfish.jersey.SslConfigurator;
-import org.glassfish.jersey.apache.connector.ApacheClientProperties;
-import org.glassfish.jersey.apache.connector.ApacheConnectorProvider;
-import org.glassfish.jersey.client.ClientProperties;
 import uk.gov.pay.commons.utils.logging.LoggingFilter;
 import uk.gov.pay.directdebit.app.config.DirectDebitConfig;
 
 import javax.inject.Inject;
-import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
 import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
 
 import static java.lang.String.format;
 
 public class ClientFactory {
-    
+
+    private static final String TLSV1_2 = "TLSv1.2";
     private final Environment environment;
     private final DirectDebitConfig conf;
 
-    private final static String PROXY_HOST_PROPERTY = "https.proxyHost";
-    private final static String PROXY_PORT_PROPERTY = "https.proxyPort";
-    
     @Inject
     public ClientFactory(Environment environment, DirectDebitConfig conf) {
         this.environment = environment;
         this.conf = conf;
     }
 
-    public Client createWithDropwizardClient(String clientName) {
-        JerseyClientConfiguration clientConfiguration = conf.getClientConfiguration();
-        JerseyClientBuilder defaultClientBuilder = new JerseyClientBuilder(environment)
-                .using(new ApacheConnectorProvider())
-                .using(clientConfiguration)
-                .withProperty(ClientProperties.READ_TIMEOUT, (int) conf.getCustomJerseyClient().getReadTimeout().toMilliseconds())
```

Comment:
We use this to set a 9 second (!) timeout via `config.yaml`

9s seems like an awfully long time, but this will presumably flip us to the dropwizard default of 500ms

Reply:
We don't have any retry capability atm, in the future we might and can reduce, and while 9 seconds is long, 500ms is probably too short?

Reply:
Looking at `dropwizard-client` it's not clear to me that the read timeout is set by dropwizard at all. It sets 'timeout' on its apache client and calls it the read timeout: https://github.com/dropwizard/dropwizard/blob/master/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java#L143

but I don't see an equivalent for the jersey client

Reply:
`HttpClientBuilder` uses the timeout value in two places:

https://github.com/dropwizard/dropwizard/blob/master/dropwizard-client/src/main/java/io/dropwizard/client/HttpClientBuilder.java#L348
and
https://github.com/dropwizard/dropwizard/blob/master/dropwizard-client/src/main/java/io/dropwizard/client/HttpClientBuilder.java#L355

Reply:
Now using Dropwizard Jersey Client which can take a `JerseyClientConfiguration` and can specify timeout there...

Reply:
It does, but I'm not convinced it configured this specific timeout. However, the way we configure the timeout currently is explicitly called out in the `dropwizard-client` documentation as unsupported and unlikely to work so I think I'm OK with this change.